lib/backend/k8s/resources/client.go

```diff
@@ -45,7 +46,7 @@ type K8sResourceClient interface {
 	// Delete removes the object specified by the KVPair.  If the KVPair
 	// contains revision information, the delete only succeeds if the
 	// revision is still current.
-	Delete(ctx context.Context, key model.Key, revision string) (*model.KVPair, error)
+	Delete(ctx context.Context, key model.Key, revision string, uid *types.UID) (*model.KVPair, error)
```

Comment:
It would be nice to remove the need for passing the `uid` explicitly here. It should probably be passed as part of the key, and respected on get/list/update/delete.

Reply:
Eh, maybe not so bad - this is only added to the k8s ResourceClient, not the main backend Client interface, so it's potentially in the right place.

The `model.Key` interface is shared across backends, and not all of them will have a UID concept, so it probably doesn't belong on the `model.Key` interface.